Microsoft.HealthVault.Fhir/Transformers/LabTestResultsToFhir.cs

```diff
@@ -0,0 +1,250 @@
+                    {
+                        Url = HealthVaultExtensions.LabTestResultValueDetail,
+                        Value = quantity
+                    };
+                    resultObservation.Value.AddExtension(HealthVaultExtensions.LabTestResultValueDetail, quantity);
+                }
+
+                foreach (var range in labTestResultValue.Ranges)
+                {
+
+                    var rangeComponent = new Observation.ReferenceRangeComponent
+                    {
+                        Type = range.RangeType.ToFhir(),
+                        Text = range.Text.Text
+                    };
+                    rangeComponent.TextElement.AddExtension(HealthVaultExtensions.LabTestResultValueRangeText, range.Text.ToFhir());
+
+                    if (range.Value != null)
+                    {
+                        rangeComponent.Low = GetSimpleQuantity(range.Value.Minimum);
+                        rangeComponent.High = GetSimpleQuantity(range.Value.Maximum);
+
+                        SimpleQuantity GetSimpleQuantity(double? value)
+                        {
+                            return value == null ? null : new SimpleQuantity() { Value = (decimal)value };
+                        }
+                    }
+
+                    resultObservation.ReferenceRange.Add(rangeComponent);
+                }
+
+                foreach (var flag in labTestResultValue.Flag)
+                {
+                    resultObservation.AddExtension(HealthVaultExtensions.LabTestResultValueFlag, flag.ToFhir());
+                }
+            }
+
+            if (resultDetail.Status != null)
+            {
+                resultObservation.Status = GetStatus(resultDetail.Status);
+            }
+
+            resultObservation.Comment = resultDetail.Note;
+        }
+
+        private static ObservationStatus? GetStatus(CodableValue status)
+        {
+            if (status == null)
+            {
+                return ObservationStatus.Unknown;
+            }
+
+            Func<CodedValue, bool> labStatusPredicate =
+                coded => coded.VocabularyName == HealthVaultVocabularies.LabStatus;
+            if (status.Any(labStatusPredicate))
+            {
+                CodedValue coded = status.First(labStatusPredicate);
+                switch (coded.Value)
+                {
+                    case HealthVaultLabStatusCodes.CompleteCode:
+                        return ObservationStatus.Final;
+                    case HealthVaultLabStatusCodes.PendingCode:
+                        return ObservationStatus.Registered;
+                    case HealthVaultLabStatusCodes.PatientRefusedTestCode:
+                        return ObservationStatus.Cancelled;
+                    case HealthVaultLabStatusCodes.QuantityNotSufficientCode:
+                        return ObservationStatus.Unknown;
+                    default:
+                        throw new ArgumentException($"Unknown Lab-status code");
```

Comment:
I feel it would be more appropriate to return `ObservationStatus.Unknown` instead of throwing since that's what they're doing if they don't have a status